enterprise/reporting/reporting_test.go

```diff
@@ -287,3 +289,372 @@ func TestReportingBasedOnConfigBackend(t *testing.T) {
+				ErrorCode:    "",
+				ErrorMessage: "",
+				EventType:    "identify",
+			},
+			ReportMetadata: types.ReportMetadata{
+				ReportedAt: 124335445,
+			},
+		},
+		{
+			PU: "dest_transformer",
+			EDInstanceDetails: types.EDInstanceDetails{
+				WorkspaceID: "wsp1",
+				InstanceID:  "instance-1",
+				Namespace:   "nmspc",
+			},
+			EDConnectionDetails: types.EDConnectionDetails{
+				SourceID:                "src-1",
+				SourceDefinitionId:      "src-def-1",
+				DestinationDefinitionId: "des-def-1",
+				DestinationID:           "des-1",
+			},
+			EDErrorDetails: types.EDErrorDetails{
+				StatusCode:   400,
+				ErrorCode:    "",
+				ErrorMessage: "bad data sent for transformation",
+				EventType:    "identify",
+			},
+			ReportMetadata: types.ReportMetadata{
+				ReportedAt: 124335445,
+			},
+		},
+		{
+			PU: "dest_transformer",
+			EDInstanceDetails: types.EDInstanceDetails{
+				WorkspaceID: "wsp1",
+				InstanceID:  "instance-1",
+				Namespace:   "nmspc",
+			},
+			EDConnectionDetails: types.EDConnectionDetails{
+				SourceID:                "src-1",
+				SourceDefinitionId:      "src-def-1",
+				DestinationDefinitionId: "des-def-1",
+				DestinationID:           "des-1",
+			},
+			EDErrorDetails: types.EDErrorDetails{
+				StatusCode:   400,
+				ErrorCode:    "",
+				ErrorMessage: "bad data sent for transformation",
+				EventType:    "identify",
+			},
+			ReportMetadata: types.ReportMetadata{
+				ReportedAt: 124335445,
+			},
+		},
+		{
+			PU: "dest_transformer",
+			EDInstanceDetails: types.EDInstanceDetails{
+				WorkspaceID: "wsp1",
+				InstanceID:  "instance-1",
+				Namespace:   "nmspc",
+			},
+			EDConnectionDetails: types.EDConnectionDetails{
+				SourceID:                "src-1",
+				SourceDefinitionId:      "src-def-1",
+				DestinationDefinitionId: "des-def-1",
+				DestinationID:           "des-1",
+			},
+			EDErrorDetails: types.EDErrorDetails{
+				StatusCode:   400,
+				ErrorCode:    "",
+				ErrorMessage: "user_id information missing",
+				EventType:    "identify",
+			},
+			ReportMetadata: types.ReportMetadata{
+				ReportedAt: 124335446,
+			},
+		},
+		// error occurred at router level(assume this is batching enabled)
+		{
+			PU: "router",
+			EDInstanceDetails: types.EDInstanceDetails{
+				WorkspaceID: "wsp1",
+				InstanceID:  "instance-1",
+				Namespace:   "nmspc",
+			},
+			EDConnectionDetails: types.EDConnectionDetails{
+				SourceID:                "src-1",
+				SourceDefinitionId:      "src-def-1",
+				DestinationDefinitionId: "des-def-1",
+				DestinationID:           "des-1",
+			},
+			EDErrorDetails: types.EDErrorDetails{
+				StatusCode:   500,
+				ErrorCode:    "",
+				ErrorMessage: "Cannot read type property of undefined", // some error during batching
+				EventType:    "identify",
+			},
+			ReportMetadata: types.ReportMetadata{
+				ReportedAt: 124335446,
+			},
+		},
+	}
+	ed := NewEdReporterFromEnvConfig()
+	reportingMetrics := ed.aggregate(dbErrs)
```

Comment:
[minor] Ideally we should not test an exported method, but instead, test the HTTP call made to reporting service.

I would adding at least one test that checks the HTTP calls

Reply:
> Ideally we should not test an exported method,

We in our tests(even before I made changes) were using non-exported methods for testing. I've also taken that approach only after looking at the already-written tests

>I would adding at least one test that checks the HTTP calls

Till now, we don't have http calls being tested in reporting. If someone from the server team can help me, write the appropriate mocks for JobsDB & any others as applicable, it would be help me in writing tests.